### DISCUSSION TITLE
Disk caching when using S3?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [lancedb/lancedb#1819] [Q&A] [answered]
If files are immutable and accessed frequently, it makes sense to have lance recognize that it could cache them locally. If you give it say 1TB nvme, it would be great for performance that we could cache files.

**Comment by eddyxu** [comment, marked as answer]
That's right. We have a NVMe cache implementation in the enterprise version of LanceDB.
https://docs.lancedb.com/enterprise/architecture/architecture

We have internal discussions about when to make it open source later.

**Reply by danthegoodman1** [reply]
Ah perfect, sorry i didn’t manage to find that in searching, thanks!
